src/main/java/pro/trevor/tankgame/rule/impl/shared/PlayerRules.java

```diff
@@ -51,9 +52,9 @@ public class PlayerRules
         new MovePositionRange("target")
     );
 
-    public static final PlayerActionRule<Tank> GetUpgradeRangeWithGoldRule(int cost)
```

Comment:
Why remove final here?

Reply:
Functions do not need to be declared final. Static + final declaration of a function is awkward.